File: PHP/UPCMatricula/views/MatriculaView.php
<?php

#require_once '../controllers/MatriculaController.php';

class MatriculaView {
    private $matriculaController;

    public function __construct() {
        $this->matriculaController = new MatriculaController();
    }

    public function matricular() {
        $opcion = '1';
        while ($opcion != '3') {
            $this->mostrarMenu();
            $opcion = $this->leerOpcion();
            $this->evaluarOpcion($opcion);
        }
    }

    public function mostrarMenu() {
        $menu = "Bienvenido al sistema\n"
              . "1 matricular alumno\n"
              . "2 listar matriculados\n"
              . "3 salir\n";
        echo $menu;
    }

    public function leerOpcion() {
        $stdin = fopen('php://stdin', 'r');
        return fgetc($stdin);
    }

    public function leerInput() {
        $stdin = fopen('php://stdin', 'r');
        return fgets($stdin, 1024);
    }

    public function evaluarOpcion($opcion) {
        if ($opcion == '1') {
            echo "Codigo: ";
            $codigo = $this->leerInput();
            echo "Nombre: ";
            $nombre = $this->leerInput();
            echo "Apellido: ";
            $apellido = $this->leerInput();
            echo $this->matriculaController->matricular($codigo, $nombre, $apellido);
        } else if ($opcion == '2') {
            echo $this->matriculaController->reporteMatricula();
        } else {
            echo "Opcion no valida";
        }
    }
}

File: Ruby/UPCMatricula/views/MatriculaView.rb
require_relative '../controllers/MatriculaController'

class MatriculaView

    def initialize
        @matricula_controller = MatriculaController.new
    end

    def matricular
        opcion = "1\n"
        while opcion != "3\n"
            self.mostrar_menu()
            opcion = self.leer_opcion()
            self.evaluar_opcion(opcion)
        end
    end

    def mostrar_menu
        menu = "Bienvenido al sistema\n"
        menu += "1 Matricular alumno\n"
        menu += "2 Listar alumno\n"
        menu += "3 Salir"
        puts menu
    end

    def leer_opcion
        return STDIN.gets()
    end

    def evaluar_opcion(opcion)
        if opcion == "1\n"
            puts "Ingrese codigo:"
            codigo = gets()
            puts "Ingrese nombre:"
            nombre = gets()
            puts "Ingrese apellido:"
            apellido = gets()
            puts @matricula_controller.matricular(codigo, nombre, apellido)
        elsif opcion == "2\n"
            puts @matricula_controller.reporte_matricula()
        else
            puts "Opcion no valida"
        end
    end

end
